lms/djangoapps/program_enrollments/signals.py

```diff
@@ -0,0 +1,47 @@
+"""
+Signal handlers for program enrollments
+"""
+import logging
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from social_django.models import UserSocialAuth
+from student.models import CourseEnrollmentException
+from third_party_auth.models import SAMLProviderConfig
+
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=UserSocialAuth)
+def populate_waiting_enrollments(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Post-save signal to update any waiting program enrollments with a user,
+    and enroll the user in any waiting course enrollments.
+
+    In most cases this will just short-circuit. Enrollments will only be updated
+    for a SAML provider with a linked organization.
+    """
+    user = instance.user
+
+    try:
+        provider_slug, external_user_key = instance.uid.split(':')
```

Comment:
is it even possible to save a model of None?  I can make sure we cover uid/user as None tho.

Reply:
#NIT Should you also handle the `None` case on `instance` passed in?